### PR TITLE
[Storybook8] Migrate Captions Settings Modal to V8

### PR DESCRIPTION
**Pull request by prabhjot-msft**
# What
Migrate Captions Settings Modal to V8
Screenshot 2024-08-19 at 4 21 10 PM

# Why

# How Tested

# Process & policy checklist

- [ ] I have updated the project documentation to reflect my changes if necessary.
- [ ] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/storybook8/stories/INTERNAL/CaptionsSettingsModal/Docs.mdx [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import * as CaptionsSettingsModalStories from './index.stories';
+import CaptionsSettingsModalExampleText from '!!raw-loader!./snippets/CaptionsSettingsModal.snippet.tsx';
+
+<Meta of={CaptionsSettingsModalStories} />
+
+# CaptionsBanner
+
+A component for displaying a CaptionsBanner with user icon, displayName and captions text.
+
+<Canvas of={CaptionsSettingsModalStories.CaptionsSettingsModal} source={{ code: CaptionsSettingsModalExampleText }} />
```
Comment (edwardlee-msft) [inline packages/storybook8/stories/INTERNAL/CaptionsSettingsModal/Docs.mdx:12]: This canvas causes the modal to pop up and take up the whole screen...
image

Comment (edwardlee-msft) [reply packages/storybook8/stories/INTERNAL/CaptionsSettingsModal/Docs.mdx:12]: Note that the modal opens up as soon as you click on the doc and covers the whole screen.

Comment (prabhjot-msft) [reply packages/storybook8/stories/INTERNAL/CaptionsSettingsModal/Docs.mdx:12]: Yeah, it is the same in storybook6 https://60c7ae6891f0e90039d7cd54-mlgfwylnxa.chromatic.com/, I think we need to add a button to open modal on click. @carocao-msft was it intended behaviour or should we modify this?

Comment (carocao-msft) [reply packages/storybook8/stories/INTERNAL/CaptionsSettingsModal/Docs.mdx:12]: We should modify this

**Review by edwardlee-msft — APPROVED** [review]

File: packages/storybook8/stories/INTERNAL/CaptionsSettingsModal/Docs.mdx (hunk 1 above)
Comment (prabhjot-msft) [reply packages/storybook8/stories/INTERNAL/CaptionsSettingsModal/Docs.mdx:12]: Done, added the button

**Review by carocao-msft — APPROVED** [review]

**Pull request merged**